components/google_sheets/sources/common/http-based/base.mjs

```diff
@@ -0,0 +1,141 @@
+    },
+    async activate() {
+      throw new Error("activate is not implemented");
+    },
+    async deactivate() {
+      const channelID = this._getChannelID();
+      const subscription = this._getSubscription();
+
+      // Reset DB state before anything else
+      this._setSubscription(null);
+      this._setChannelID(null);
+
+      await this.googleSheets.deactivateHook(channelID, subscription?.resourceId);
+    },
+  },
+  methods: {
+    _getSubscription() {
+      return this.db.get("subscription");
+    },
+    _setSubscription(subscription) {
+      this.db.set("subscription", subscription);
+    },
+    _getChannelID() {
+      return this.db.get("channelID");
+    },
+    _setChannelID(channelID) {
+      this.db.set("channelID", channelID);
+    },
+    async getAllWorksheetIds(sheetID) {
+      const { sheets } = await this.googleSheets.getSpreadsheet(sheetID);
+      return sheets
+        .map(({ properties }) => properties)
+        .filter(({ sheetType }) => sheetType === "GRID")
+        .map(({ sheetId }) => (sheetId.toString()));
+    },
+    getDriveId(drive = this.watchedDrive) {
+      return googleSheets.methods.getDriveId(drive || MY_DRIVE_VALUE);
+    },
+    getSheetId() {
+      throw new Error("getSheetId is not implemented");
+    },
+    getWorksheetIds() {
+      throw new Error("getWorksheetIds is not implemented");
+    },
+    isEventRelevant(event) {
+      const { headers } = event;
+      return headers["x-goog-resource-state"] !== "sync";
+    },
+    isWorksheetRelevant(worksheetId) {
+      const worksheetIds = this.getWorksheetIds();
+      return worksheetIds.includes(worksheetId.toString());
+    },
+    processSpreadsheet() {
+      throw new Error("processEvent is not implemented");
+    },
+    async renewSubscription() {
+      throw new Error("renewSubscription is not implemented");
+    },
+    /**
+     * This method scans the worksheets indicated by the user to retrieve the
+     * current row count of each one, and cache those values.
+     * @param {number} [offset=0] When present, the row count that gets cached
+     * will be reduced by this amount (useful for example to force an event
+     * source to interpret the last rows as new).
+     */
+    takeSheetSnapshot() {
+      throw new Error("takeSheetSnapshot is not implemented");
+    },
+  },
+  async run(event) {
+    if (event.timestamp) {
+      // Component was invoked by timer
+      return this.renewSubscription();
+    }
+
+    if (!this.isEventRelevant(event)) {
+      console.log("Sync notification, exiting early");
+      return;
+    }
+
+    const spreadsheet = this.isMyDrive()
+      ? await this.googleSheets.getSpreadsheet(this.sheetID)
+      : await this.getSpreadsheetToProcess(event);
+
+    if (!spreadsheet) {
+      const sheetId = this.getSheetId();
+      console.log(`Spreadsheet "${sheetId}" was not modified. Skipping event`);
+      return;
+    }
+
+    return this.processSpreadsheet(spreadsheet);
+  },
```

Comment:
Isn't this overwritten in both drive.mjs and sheet.mjs? I think it could be left out if it isn't being used anywhere.
```suggestion
```